thunder/tests/test_update_aliases.py

```diff
@@ -70,6 +79,62 @@ def inplace_masked_fill_sample_generator(op, device, dtype, requires_grad, **kwa
         _inplace_opinfos.append(inplace_opinfo)
 
 
+@pytest.fixture
+def turn_off_tf32_and_set_seed(monkeypatch):
+    import torch
+
+    monkeypatch.setenv("NVIDIA_TF32_OVERRIDE", "0")
+    torch.manual_seed(42)
+
+
+@instantiate(
+    dtypes=(thunder.float32, thunder.float64),
+    devicetypes=(devices.DeviceType.CUDA,),
+    decorators=(pytest.mark.parametrize("train", (False, True)),),
+)
+@requiresCUDA
+def test_parse_resnet18(executor, device, dtype, turn_off_tf32_and_set_seed, train: bool):
+    from contextlib import nullcontext
+
+    print(executor.name)
+    import thunder
+
+    torchvision = pytest.importorskip("torchvision")
+
+    tdtype = thunder.torch.to_torch_dtype(dtype)
+    model = torchvision.models.resnet18(weights=None).to(device=device, dtype=tdtype)
+    ref_model = torchvision.models.resnet18(weights=None).to(device=device, dtype=tdtype)
+    if not train:
+        model = model.eval()
+        ref_model = ref_model.eval()
+        ctx = torch.no_grad
+    else:
+        model = model.train()
+        ref_model = ref_model.train()
+        ctx = nullcontext
+    ref_model.load_state_dict(model.state_dict())
+
+    jitted = executor.make_callable(model, skip_inplace_alias_updates=True, skip_inplace_functionalization=False)
+    x = make_tensor((1, 3, 224, 224), dtype=tdtype, device=device)
+
+    with ctx():
+        out1 = ref_model(x)
+        out2 = jitted(x)
+        torch.testing.assert_close(out1, out2, atol=1e-4, rtol=1e-1)
+        # Numerical accuracy error when TorchExecutor, `train=True` and dtype is fp32.
+        # with RTX6000 Ada and CUDA 12.3, I see somewhat huge error:
+        # E   AssertionError: Tensor-likes are not close!
+        # E
+        # E   Mismatched elements: 9401 / 9408 (99.9%)
+        # E   Greatest absolute difference: 0.07035164535045624 at index (4, 1, 0, 3) (up to 1e-05 allowed)
+        # E   Greatest relative difference: 343.7076110839844 at index (5, 0, 5, 4) (up to 1.3e-06 allowed)
+        # E   The failure occurred for item [0]
```

Comment:
Could you double check this if? My bad I never revisited it but cuda 12.3 reads a bit old

Reply:
I did not see errors as huge as noted in the comment, but still enough that changing the tolerances is necessary.